app/services/search/search_builder.rb

```diff
@@ -2,62 +2,65 @@ class Search::SearchBuilder
   DEFAULT_HITS_PER_PAGE = 10
   DEFAULT_PAGE = 1
 
-  attr_reader :hits_per_page, :keyword, :location_search, :page, :params_hash,
-              :point_coordinates, :search_filters, :search_replica, :sort_by, :stats, :total_count, :vacancies,
-              :wider_search_suggestions
+  attr_reader :params_hash, :keyword, :page, :hits_per_page, :stats, :total_count, :vacancies
 
   def initialize(form_hash)
     @params_hash = form_hash
     @keyword = @params_hash[:keyword]
     @hits_per_page = (@params_hash[:per_page] || DEFAULT_HITS_PER_PAGE).to_i
     @page = (@params_hash[:page] || DEFAULT_PAGE).to_i
 
-    build_location_search
-    build_search_filters
-    build_search_replica
     call_search
-    build_suggestions if @vacancies.empty? && any?
   end
 
   def only_active_to_hash
-    @params_hash.merge(keyword: @keyword).delete_if do |k, v|
-      v.blank? || (k.eql?(:radius) && @params_hash[:location].blank?) || k.eql?(:jobs_sort) || k.eql?(:page)
+    @params_hash.merge(keyword: @keyword).except(:jobs_sort, :page).reject do |k, v|
+      v.blank? || (k.eql?(:radius) && @params_hash[:location].blank?)
```

Comment:
Instead of `v.blank?`:

https://ruby-doc.org/core-2.4.0/Hash.html#method-i-compact

Reply:
Good shout!

Reply:
I take that back - we need to explicitly take blank (e.g. empty string) into account too here!

Reply:
Oh!